### PR TITLE
Ported twisted.python._release to Python 3

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/8566

### CODE REVIEW

File: src/twisted/python/test/test_release.py [hunk 1]
```diff
@@ -861,7 +861,7 @@ def test_statusNotClean(self):
         no pending modifications returns C{False}.
         """
         reposDir = self.makeRepository(self.tmpDir)
-        reposDir.child('some-file').setContent("something")
+        reposDir.child('some-file').setContent(b"something")
```
Comment (acabhishek942) [inline src/twisted/python/test/test_release.py:864]: is it not required to make `'some-file'` as bytes?

File: src/twisted/python/test/test_release.py [hunk 2]
```diff
@@ -872,7 +872,7 @@ def test_remove(self):
         """
         reposDir = self.makeRepository(self.tmpDir)
         testFile = reposDir.child('some-file')
```
Comment (acabhishek942) [inline src/twisted/python/test/test_release.py:874]: same here

File: src/twisted/python/_release.py [hunk 3]
```diff
@@ -455,16 +456,16 @@ def _writeSection(self, fileObj, header, tickets):
             reverse[description].sort()
         reverse = reverse.items()
         # result is a tuple of (descr, tickets)
-        reverse.sort(key=lambda result: result[1][0])
+        reverse = sorted(reverse, key=lambda result: result[1][0])
```
Comment (acabhishek942) [inline src/twisted/python/_release.py]: why is this change necessary?

File: src/twisted/python/_release.py [hunk 4]
```diff
@@ -455,16 +456,16 @@ def _writeSection(self, fileObj, header, tickets):
...
 
-        fileObj.write(header + '\n' + '-' * len(header) + '\n')
+        fileObj.write((header + u'\n' + u'-' * len(header) + u'\n').encode("utf-8"))
         for (description, relatedTickets) in reverse:
             ticketList = ', '.join([
                 '#' + str(ticket) for ticket in relatedTickets])
             entry = ' - %s (%s)' % (description, ticketList)
```
Comment (acabhishek942) [inline src/twisted/python/_release.py]: I think it would be better to use `format` method here and in other applicable places to keep the code consistent

File: src/twisted/python/test/test_release.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/python/test/test_release.py:864]: No it is not required to make ```'some-file'``` bytes.  Only the argument for ```setContent()``` must be bytes.

File: src/twisted/python/_release.py (hunk 3 above)
Comment (rodrigc) [reply src/twisted/python/_release.py]: This code has been deleted, so this change is no longer there.

**Comment by acabhishek942** [comment]
Please update the patch. Automated tests are not hitting all the lines

**Comment by rodrigc** [comment]
Coverage is not reaching all the lines because sphinx is not installed on the builders where coverage is statistics are uploaded to codecov.  I mentioned the same thing here: https://github.com/twisted/twisted/pull/713#discussion_r104746647

**Comment by acabhishek942** [comment]
> Coverage is not reaching all the lines because sphinx is not installed on the builders where coverage is statistics are uploaded to codecov. I mentioned the same thing here: #713 (comment)

In this case the patch looks good to me. Please merge once all the builds are green
